Bottom: be1dc7d79cd27a817e87bb44aa79287e7fbd68ad
Top:    31afbc3b578d3df9f9bb440bd0defa44c47b6a4d
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-12-24 17:24:55 +0000

hid/gtk: Move ghid_gui_is_up variable into the ghidgui structure.

This lets us access it outside of gtkhid-main.c


---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index db7d831..eb0ad14 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -239,12 +239,10 @@ ghid_calibrate (double xval, double yval)
   printf (_("ghid_calibrate() -- not implemented\n"));
 }
 
-static int ghid_gui_is_up = 0;
-
 void
 ghid_notify_gui_is_up ()
 {
-  ghid_gui_is_up = 1;
+  ghidgui->is_up = 1;
 }
 
 int
@@ -253,7 +251,7 @@ ghid_shift_is_pressed ()
   GdkModifierType mask;
   GHidPort *out = &ghid_port;
 
-  if( ! ghid_gui_is_up ) 
+  if (!ghidgui->is_up)
     return 0;
 
   gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
@@ -267,7 +265,7 @@ ghid_control_is_pressed ()
   GdkModifierType mask;
   GHidPort *out = &ghid_port;
 
-  if( ! ghid_gui_is_up )
+  if (!ghidgui->is_up)
     return 0;
 
   gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
@@ -281,7 +279,7 @@ ghid_mod1_is_pressed ()
   GdkModifierType mask;
   GHidPort *out = &ghid_port;
 
-  if( ! ghid_gui_is_up )
+  if (!ghidgui->is_up)
     return 0;
 
   gdk_window_get_pointer (gtk_widget_get_window (out->drawing_area),
diff --git a/src/hid/gtk/gui.h b/src/hid/gtk/gui.h
index e4d55dd..34b4c11 100644
--- a/src/hid/gtk/gui.h
+++ b/src/hid/gtk/gui.h
@@ -146,6 +146,8 @@ typedef struct
     library_window_width,
     library_window_height,
     netlist_window_height, history_size, settings_mode;
+
+  bool is_up;
 }
 GhidGui;
